src/validations/javascript/jshint.js

```diff
@@ -139,11 +151,11 @@ const errorMap = {
 };
 
 class JsHintValidator extends Validator {
-  constructor(source, enabledLibraries) {
-    super(source, 'javascript', errorMap);
-
-    this._jshintOptions = defaults(clone(jshintrc), {predef: []});
-    enabledLibraries.forEach((libraryKey) => {
+  constructor(source, analyzer) {
+    super(source, 'javascript', errorMap, analyzer);
+    const jshintConfig = this._getConfig(analyzer);
+    this._jshintOptions = defaults(clone(jshintConfig), {predef: []});
+    analyzer.enabledLibraries.forEach((libraryKey) => {
       if (!(libraryKey in libraries)) {
         return;
       }
```

Comment:
Ok, now `_getConfig` calls through to two helpers, one for figuring out jshint configuration and one for mutating based on enabled libraries.